src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -1250,14 +1250,73 @@ CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::PointIsValid(void * R)
     return CHIP_NO_ERROR;
 }
 
+namespace {
+
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
-static int mbedTLSCallbackIgnoreCertValidityCheck(void * data, mbedtls_x509_crt * crt, int depth, uint32_t * flags)
+bool IsTimeGreaterThanEqual(const mbedtls_x509_time * const timeA, const mbedtls_x509_time * const timeB)
+{
+
+    // checks if two values are different and if yes, then returns first > second.
+#define RETURN_STRICTLY_GREATER_IF_DIFFERENT(component)                                                                            \
+    {                                                                                                                              \
+        auto valueA = timeA->CHIP_CRYPTO_PAL_PRIVATE_X509(component);                                                              \
+        auto valueB = timeB->CHIP_CRYPTO_PAL_PRIVATE_X509(component);                                                              \
+                                                                                                                                   \
+        if (valueA != valueB)                                                                                                      \
+        {                                                                                                                          \
+            return valueA > valueB;                                                                                                \
+        }                                                                                                                          \
+    }
+
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(year);
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(mon);
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(day);
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(hour);
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(min);
+    RETURN_STRICTLY_GREATER_IF_DIFFERENT(sec);
+
+    // all above are equal
+    return true;
+}
+
+CHIP_ERROR IsCertificateValidAtIssuance(const mbedtls_x509_crt * referenceCertificate,
+                                        const mbedtls_x509_crt * toBeEvaluatedCertificate)
```

Comment:
This API is used as follows
```
if (IsCertificateValidAtIssuance(leafCert, tbeCert) != CHIP_NO_ERROR)
```

Suggest renaming arguments so that `toBeEvaluedCertificate` -> `issuerCertificate` and `referenceCertificate` -> `candidateCertificate`. Otherwise the naming makes it appear that `referenceCertificate` (which is the leaf cert that must be checked against issuer timing) is the one that has the bounds to check against, when  it's actually the certificate being tested.

Reply:
updated naming